flamingpy/utils/viz.py

```diff
@@ -561,27 +571,14 @@ def syndrome_plot(code, ec, index_dict=None, drawing_opts=None):
                     # fontdict=font_props
                 )
 
-    # This portion adapted from a Matplotlib official example to fix
-    # an issue with filling in the insides of voxels: the code
-    # expands the indices and creates small gaps between the voxels.
-
-    def explode(data):
-        size = np.array(data.shape) * 2
-        data_e = np.zeros(size - 1, dtype=data.dtype)
-        data_e[::2, ::2, ::2] = data
-        return data_e
-
-    # upscale the above voxel image, leaving gaps
-    filled_e = explode(filled)
-    # Shrink the gaps
-    x, y, z = np.indices(np.array(filled_e.shape) + 1, dtype=float)
-    x[0::2, :, :] += 0.05
-    y[:, 0::2, :] += 0.05
-    z[:, :, 0::2] += 0.05
-    x[1::2, :, :] += 0.95
-    y[:, 1::2, :] += 0.95
-    z[:, :, 1::2] += 0.95
-    ax.voxels(x, y, z, filled_e, facecolors=filled_e)
+    # draw cubes
+    pc = _plot_cube_at(positions, colors=colors, sizes=sizes)
+    ax.add_collection3d(pc)
+
+    # setting plot limits to give some room to the boxes
+    ax.set_xlim(-2, 2 * shape[0] + 1)
+    ax.set_ylim(-2, 2 * shape[1] + 1)
+    ax.set_zlim(-2, 2 * shape[2] + 1)
```

Comment:
```suggestion
    ax.set_xlim(-1, 2 * shape[0])
    ax.set_ylim(-1, 2 * shape[1])
    ax.set_zlim(-1, 2 * shape[2])
```
I think this works a little better, but please confirm it's not too tight.